docs/source/native/class.rst

```diff
@@ -0,0 +1,186 @@
+Classes
+=======
+
+.. note::
+
+    The version Respawn is using differs in some places from classes that are in use in Squirrel 3.x 
+
+    This is by no means complete. Please add anything you know.
+
+Declaring Classes
+-----------------
+
+To declare a class, first add the ``untyped`` keyword and the class as a global variable at file level.
+
+.. code-block:: javascript
+
+    untyped
+    global var ExampleClass
+
+The ``untyped`` declaration is required because instances have an unknown type and it's not possible to use classes as types.
+
+``global var [classname]`` represents the class. After declaring the class inside of a function you can use it anywhere. It's not possible to declare classes on local variables.
+
+You can then add the declaration inside of a function.
+
+Most classes use a constructor. A constructor is a function of the instance that gets executed on object creation.
+
+.. code-block:: javascript
+
+    void function initClient() {
+        class ExampleClass {
+            constructor(){print("Instance of ExampleClass created");}
+        }
+    }
+
+You can require parameters in the constructor. Keep in mind that you have to pass those when creating an object.
+
+The types of the parameters have to be ``var``, but you can ``expect`` types in the constructor as usual.
+
+.. code-block:: javascript
+
+    class ExampleClass {
+            propertyString // Actual type is var
+            propertyInt // Actual type is var
```

Comment:
the way you document properties is wrong, they have to be ` = null` (or = default) ![](https://i.laundmo.com/tENe0/bODolUdU11.png/raw)